File: app/src/main/java/com/kidssaveocean/fatechanger/onboarding/OnboardingFragment.java
package com.kidssaveocean.fatechanger.onboarding;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.constraint.ConstraintLayout;
import android.support.v4.app.Fragment;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.kidssaveocean.fatechanger.R;
import com.kidssaveocean.fatechanger.bottomNavigation.BottomNavigationActivity;
import com.kidssaveocean.fatechanger.userIdentification.UserIdentificationActivity;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Optional;

public class OnboardingFragment extends Fragment {

    public static final String IMAGE = "image";
    public static final String TITLE = "title";
    public static final String SUBTITLE = "subtitle";
    public static final String DESCRIPTION = "description";
    public static final String LAYOUT = "layout";

    @BindView(R.id.onboarding_image) ImageView onboardingImage;
    @BindView(R.id.title) TextView titleTextView;
    @BindView(R.id.subtitle) TextView subtitleTextView;
    @BindView(R.id.description) TextView descriptionTextView;

    public OnboardingFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        Bundle bundle = getArguments();
        int imageId = bundle.getInt(IMAGE);
        int titleTextId = bundle.getInt(TITLE);
        int subTitleTextId = bundle.getInt(SUBTITLE);
        int descriptionTextId = bundle.getInt(DESCRIPTION);
        int layoutId = bundle.getInt(LAYOUT);

        ConstraintLayout layout = (ConstraintLayout) inflater.inflate(layoutId, container, false);
        ButterKnife.bind(this, layout);

        onboardingImage.setImageDrawable(getResources().getDrawable(imageId));
        titleTextView.setText(titleTextId);
        subtitleTextView.setText(subTitleTextId);
        descriptionTextView.setText(descriptionTextId);

        return layout;
    }

    @Optional
    @OnClick(R.id.start_button)
    public void clickStart (Button startButton) {
        Intent intent = new Intent(getActivity(), BottomNavigationActivity.class);
        startActivity(intent);
        return;
    }


}

File: app/src/main/java/com/kidssaveocean/fatechanger/onboarding/OnboardingFragment.kt
package com.kidssaveocean.fatechanger.onboarding

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.ScrollView
import android.widget.TextView
import androidx.fragment.app.Fragment
import com.kidssaveocean.fatechanger.R
import com.kidssaveocean.fatechanger.onboarding.userIdentification.UserIdentificationActivity

//todo what is this abomination ?????
class OnboardingFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val bundle = arguments
        val imageId = bundle!!.getInt(IMAGE)
        val titleTextId = bundle.getInt(TITLE)
        val subTitleTextId = bundle.getInt(SUBTITLE)
        val descriptionTextId = bundle.getInt(DESCRIPTION)
        val layoutId = bundle.getInt(LAYOUT)
        val extraId = bundle.getInt(EXTRA)

        val layout = inflater.inflate(layoutId, container, false) as ScrollView

        layout.findViewById<ImageView>(R.id.onboarding_image).setImageDrawable(resources.getDrawable(imageId))
        layout.findViewById<TextView>(R.id.title).setText(titleTextId)
        layout.findViewById<TextView>(R.id.subtitle).setText(subTitleTextId)
        layout.findViewById<TextView>(R.id.description).setText(descriptionTextId)
        layout.findViewById<TextView>(R.id.extra).setText(extraId)

        layout.findViewById<Button?>(R.id.start_button)?.setOnClickListener { view ->
            val intent = Intent(activity, UserIdentificationActivity::class.java)
            startActivity(intent)
        }

        return layout
    }

    companion object {

        val IMAGE = "image"
        val TITLE = "title"
        val SUBTITLE = "subtitle"
        val DESCRIPTION = "description"
        val LAYOUT = "layout"
        val EXTRA = "extra"
    }


}
